pytest/lib/cluster.py

```diff
@@ -398,24 +398,25 @@ def start(self, *, boot_node: BootNode = None, skip_starting_proxy=False):
         except:
             logger.error(
                 '=== failed to start node, rpc does not ready in 10 seconds')
-            self.stdout.close()
-            self.stderr.close()
             if os.environ.get('BUILDKITE'):
                 logger.info('=== stdout: ')
-                logger.info(open(self.stdout_name).read())
+                self._stdout.seek(0)
+                logger.info(self._stdout.read())
                 logger.info('=== stderr: ')
-                logger.info(open(self.stderr_name).read())
+                self._stderr.seek(0)
+                logger.info(self._stdout.read())
+            self._stdout.close()
+            self._stderr.close()
+            self._stdout = None
+            self._stderr = None
 
     def kill(self):
-        if self.pid.value != 0:
-            try:
-                os.kill(self.pid.value, signal.SIGKILL)
-            except ProcessLookupError:
-                pass  # the process has already terminated
-            self.pid.value = 0
-
-            if self._proxy_local_stopped is not None:
-                self._proxy_local_stopped.value = 1
+        if self._proxy_local_stopped is not None:
+            self._proxy_local_stopped.value = 1
+        if self._process:
+            self._process.kill()
+            self._process.wait(5)
```

Comment:
this will add quite a delay to all the tests (and might introduce some flakiness). maybe we could keep checking that pid is free?

Reply:
I don’t see why it would. The process is killed with SIGKILL so in practice this will never wait.